src/data/viteTrickFetcher.ts

```diff
@@ -0,0 +1,158 @@
+/**
+ * **WARNING** This module is expected to be invoked from a Vite Plugin!
+ * 
+ * Aggregated all YAML Files in the ./tricks Directory and creates a list 
+ */
+
+
+// unfortunately, vite features (e.g. import.meta.glob) are NOT available to plugin code.
+// as such, we must fall back to reading YAMLs manually.
+import {globby} from "globby"
+import {join} from "path";
+import {fileURLToPath} from "url"
+import { ZodError, z } from "zod";
+import { DbTricksTableZod } from "../lib/database/schemas/CurrentVersionSchema"
+import {parse} from "yaml"
+import { readFile } from "fs/promises";
+import { YamlTrickTableSchemaZod } from "./viteSchemaGenerator";
+import buildGraph from "./variationOfCycleChecker";
+
+async function fetchYamlFile(path: string): Promise<z.infer<typeof DbTricksTableZod>> {
+  const asObject = YamlTrickTableSchemaZod.parse(parse(await readFile(path, "utf8")))
+  if(asObject.recommendedPrerequisites) {
+    console.log(asObject.recommendedPrerequisites)
+    asObject.recommendedPrerequisites = asObject.recommendedPrerequisites.map( prerequisiteId => ([prerequisiteId, "official"])) as any
+  }
+  if(asObject.variationOf) {
+    asObject.variationOf = asObject.variationOf.map( variationId => ([variationId, "official"])) as any
+    
+  }
+  // Makes sure the YAML has the right structure. 
+  return DbTricksTableZod.parse({...asObject, trickStatus: "official"})
+}
+
+
+function findDuplicateKeys(allTricks: z.infer<typeof DbTricksTableZod>[]) {
+
+  let idToTrickLookup: Record<number, (z.infer<typeof DbTricksTableZod>)[]> = createRecordLookup(allTricks);
+
+  return Object.entries(idToTrickLookup).filter(([_,trickList]) => trickList.length > 1).map(([id,trickList]) => ({
+    id,
+    technicalNames: trickList.map( trick => trick.technicalName)
+  }))
+
+}
+
+
+function createRecordLookup(allTricks: z.infer<typeof DbTricksTableZod>[]) {
+  let idToTrickLookup: Record<number, (z.infer<typeof DbTricksTableZod>)[]> = {};
+
+  allTricks.forEach(trick => {
+    if (!idToTrickLookup[trick.id]) {
+      idToTrickLookup[trick.id] = [];
+    }
+      idToTrickLookup[trick.id].push(trick);
+    });
+  return idToTrickLookup;
+}
+
+/**
+ * Tricks contain two types of References: 
+ * - recommendedPrerequisites
+ * - variationOf
+ */
+function findUndefinedReferences(allTricks: z.infer<typeof DbTricksTableZod>[]) {
+  
+  let idToTrickLookup: Record<number, (z.infer<typeof DbTricksTableZod>)[]> = createRecordLookup(allTricks);
+
+  // issues define the ID of the Trick and an issue message
+  const issues: [number, string][] = []
+
+  allTricks.forEach( trick => {
+    if(trick.recommendedPrerequisites) {
+      trick.recommendedPrerequisites.forEach( (prerequisitePrimaryKey,i) => {
+        if(prerequisitePrimaryKey[1] !== "official") {
```

Comment:
As in the comboFetcher: Is it possible to use dot notation here?

Reply:
Nope. `.forEach` iterates over any prerequisites. Do note however, that `trick.recommendedPrerequisites` does not contain another Trick object, but a Tuple defining the Primary Key.